specs/super_circuit.md

```diff
@@ -0,0 +1,39 @@
+# Super Circuit
+
+The zkEVM is composed of many circuits, each of which is responsible for checking some EVM aspect.
+These circuits need to be aggregated or combined together somehow. We could say that the Super
+Circuit is one of the possible strategies to combine all other sub-circuits.
+
+The Super Circuit approach is quite simple, all sub-circuits are put together and their tables are
+shared among all of them. It contains the following circuits:
+ - [x] EVM Circuit
+ - [x] State Circuit
+ - [ ] MPT Circuit
+ - [x] Keccak Circuit
+ - [x] Tx Circuit
+ - [x] Bytecode Circuit
+ - [x] Copy Circuit
+ - [ ] Block Circuit
+ - [x] PublicInputs Circuit
+
+ and the following tables:
+ - [x] RW Table
+ - [x] MPT Table
+ - [x] Keccak Table
+ - [x] Tx Table
+ - [x] Bytecode Table
+ - [x] Copy Table
+ - [x] Block Table
+
+The following diagram shows the relation between this circuits and tables:
+![](./super_circuit.png)
+
+**Note**: There are some smaller sub-circuits, like the ECDSA circuit and opcode related
```

Comment:
I think the tx circuit should look up signatures in ecdsa circuit table to verify signature's correctness.

Reply:
Exactly; currently the TxCircuit uses halo2wrong to verify correct ECDSA signatures, and the connection is made via copy constraints.

AFAIK the only signature verification needed are for the transactions and for the `ecrecover` precompile.  Once we implement the precompile we may have an ECDSA signature verification circuit that exposes a table where other circuits can look up; but it's not clear yet if we'll choose this direction.  A detail about `ecrecover` that is different than the signature verification of transactions is that it must support invalid signatures.